pkgs/applications/virtualization/kvmtool/default.nix

```diff
@@ -0,0 +1,21 @@
+{ stdenv, fetchgit }:
+
+stdenv.mkDerivation rec {
+  pname = "kvmtool";
+  version = "0.20200424";
+
+  src = fetchgit {
+    url = "git://git.kernel.org/pub/scm/linux/kernel/git/will/${pname}.git";
+    rev = "c0c45eed4f3fb799764979dec5cfb399071d6916";
+    sha256 = "0ir6aqvipss145r85324nw84k4ilzxa2rsa1wr5303p1h6gs0qkk";
+  };
+
+  makeFlags = [ "DESTDIR=$(out)" "prefix=/" ];
```

Comment:
I assume it tries to install something outside of `prefix`? We usually we should avoid `prefix=` as it can lead to wrong references within the package.

Reply:
It doesn't try to install outside of prefix.  It oddly uses `$HOME` as its default `prefix`, which causes a `homeless-shelter` path component in the installed path.  Default `DESTDIR` is empty.

Oh, I see that I don't need both DESTDIR and prefix -- just `prefix=$(out)` also works.

| makeFlags       | Outcome |
| ---- | ---- |
| (nothing)          | `cannot create directory '/homeless-shelter': Permission denied` |
| `DESTDIR=$(out)` | Executable is `result/homeless-shelter/bin/lkvm` (bad) |
| `DESTDIR=$(out) prefix=/` | Executable is `result/bin/lkvm` (good) |
| `DESTDIR=$(out) prefix=` | Executable is `result/bin/lkvm` (good) |
|  `prefix=$(out)` | Executable is `result/bin/lkvm` (good) |

All three of the entries labelled "(good)" produce the same binary (hashes to 3c896d01...).

Which is preferred?

Reply:
The latter one: `prefix=$(out)`.